admin-api-server/src/nft/service/file/file.service.ts

```diff
@@ -0,0 +1,383 @@
+import { Injectable, Logger } from '@nestjs/common';
+import fs from 'fs';
+import { Buffer } from 'node:buffer';
+import ffmpeg from 'fluent-ffmpeg';
+import { fileURLToPath } from 'url';
+import * as path from 'path';
+import sharp from 'sharp';
+import sizeOf from 'image-size';
+import mime from 'mime';
+import { getContentMetadata, File } from '../../../media.js';
+
+mime.define(
+  {
+    'video/mov': ['mov'],
+    'video/quicktime': ['qt'],
+  },
+  true,
+);
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+
+type FileType = 'thumbnail' | 'display';
+type Path = string;
+export type Dimension = {
+  width: number;
+  height: number;
+};
+
+interface ResizedImgParams {
+  buffer: Buffer;
+  width: number;
+  height: number;
+  type: FileType;
+  extension: string;
+}
+
+interface ImageFromVideoParams {
+  tempDir: string;
+  videoFile: File;
+  typeOfImage: FileType;
+  imageWidth: number;
+  imageHeight: number;
+}
+
+interface GenerateImageFileParams {
+  source: Path | Buffer;
+  type: FileType;
+  extension: string;
+}
+
+interface ScreenshotFromVideoParams {
+  tempDir: string;
+  imageWidth: number;
+  imageHeight: number;
+  videoPath: string;
+  filename: string;
+}
+
+interface HandleMediaParams {
+  nftId: number;
+  tempDir: string;
+  artifact: File;
+  display?: File;
+  thumbnail?: File;
+}
+
+interface AddMissingFiles {
+  filesArray: File[];
+  tempDirId: string;
+  nftId: number;
+}
+
+@Injectable()
+export class FileService {
+  async addMissingFiles({
+    filesArray,
+    tempDirId,
+    nftId,
+  }: AddMissingFiles): Promise<File[]> {
+    const MEDIA_TEMP_DIR = __dirname + `/${tempDirId}/`;
```

Comment:
all capital letters should be reserved for constants, this isn't a constant, it's something that varies over different addMissingFiles executions.

Reply:
Adjusted to camel case